### PR TITLE
Gallery video upload

### PR DESCRIPTION
**Pull request by bbruil**
Issue#78 adding in place to upload 4 GalleryVideos for the AthleteIndexPage.

### CODE REVIEW

**Comment by bbruil** [comment]
This also contains the Athlete Index page changes for issue #55

**Review by adamkohler — APPROVED** [review]

File: gybitg/Migrations/20190531010057_GalleryVideos model update.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace gybitg.Migrations
+{
+    public partial class GalleryVideosmodelupdate : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.AddColumn<string>(
+                name: "GalleryVideo1",
```
Comment (Bansenauer-Cascadia) [inline gybitg/Migrations/20190531010057_GalleryVideos model update.cs:10]: Brenda, listing out individual GalleryVideo columns in the data tables will be a problem as the application grows; where you will have to change the database structure to add a new column. Instead, create a many-to-many relationship that can continue to add as many records as needed.

File: gybitg/Models/ManageViewModels/AthleteUserViewModel.cs [hunk 2]
```diff
@@ -33,8 +37,51 @@ public virtual string FullName
 
         public string ProfileVideoUrl { get; set; }
 
-        [DisplayFormat(DataFormatString = "{0:yyyy-MM}", ApplyFormatInEditMode = true)]
+        public string GalleryVideo1 { get; set; }
```
Comment (Bansenauer-Cascadia) [inline gybitg/Models/ManageViewModels/AthleteUserViewModel.cs:40]: Refactor listing like this of repeated properties as an array of list

File: gybitg/Views/Athlete/Index.cshtml [hunk 3]
```diff
@@ -1,77 +1,234 @@
...
+
+</div>
+
+
+<br />
+<h1 style="text-align:center"> GALLERY </h1>
+<br />
+<div class="col-md-4">
+
+    <div class="video-container-gallery">
```
Comment (Bansenauer-Cascadia) [inline gybitg/Views/Athlete/Index.cshtml:208]: Refactor as a for-each loop through your GalleryVideo collection

**Review by Bansenauer-Cascadia — COMMENTED** [review]
Please see some refactoring comments regarding the GalleryVideo data and views.

File: gybitg/Migrations/20190531010057_GalleryVideos model update.cs (hunk 1 above)
Comment (bbruil) [reply gybitg/Migrations/20190531010057_GalleryVideos model update.cs:10]: @Bansenauer-Cascadia ah yes that makes sense. :)

File: gybitg/Views/Athlete/Index.cshtml (hunk 3 above)
Comment (bbruil) [reply gybitg/Views/Athlete/Index.cshtml:208]: Yes that makes sense.

File: gybitg/Migrations/20190531010057_GalleryVideos model update.cs (hunk 1 above)
Comment (bbruil) [reply gybitg/Migrations/20190531010057_GalleryVideos model update.cs:10]: I will see how far I can get in refactoring this tonight with the last of my hours. :)

**Comment by bbruil** [comment]
I am going to pull the latest develop branch (after the merges that happened today) and I will merge that locally with my 05-20sprint branch that was pre video gallery. I will then create a new pull request.

**Pull request closed without merging**